### PR TITLE
Add way to get resolved dependencies

### PR DESCRIPTION
**Pull request by jaredoconnell**
## Changes introduced with this PR

This PR adds a way to keep track of dependencies that were resolved.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: dg.go [hunk 1]
```diff
@@ -438,6 +447,9 @@ func (n *node[NodeType]) dependencyResolved(dependencyNodeID string, dependencyR
 			panic(ErrConnectionDoesNotExist{dependencyNodeID, n.id})
 		}
 	}
+	if dependencyResolution == Resolved {
+		n.resolvedDependencies[dependencyNodeID] = dependencyType
```
Comment (dbutenhof) [inline dg.go:451]: If you're locking when retrieving the map in `ResolvedDependencies`, don't you want to lock for consistency when *changing* the map?

Comment (jaredoconnell) [reply dg.go:451]: The lock is supposed to be acquired for the entirety of that function's scope. So it's locked there.

Comment (dbutenhof) [reply dg.go:451]: OK, I'll trust you rather than searching up the call tree. 😆

**Review by dbutenhof — APPROVED** [review]

File: dg.go (hunk 1 above)
Comment (webbnh) [reply dg.go:451]: You don't have to search the call tree -- just look at the function header comment:  it says you have to have the lock when you call the function, so I'm sure we're covered.  😉

**Review by mfleader — APPROVED** [review]

**Pull request merged**